oneview_redfish_toolkit/blueprints/subscription.py

```diff
@@ -122,6 +122,34 @@ def add_subscription():
         return abort(status.HTTP_500_INTERNAL_SERVER_ERROR)
 
 
+@subscription.route(
+    "/redfish/v1/EventService/EventSubscriptions/<id>", methods=["DELETE"])
+def remove_subscription(id):
+    """Removes a specific Subscription
+
+        Args:
+            id: The Subscription ID.
+
+        Returns:
+            Subscription JSON.
+    """
+    try:
+        sc = util.all_subscriptions[id]
+        event_types = sc.redfish["EventTypes"]
+
+        for event in event_types:
+            del util.subscriptions_by_type[event][id]
```

Comment:
This line also can yield a KeyError, right?

I know it is a corner case... but right now it is impossible to delete a subscription created with:

```
{
  "Destination": "http://www.myserveruri.com",
  "EventTypes": ["Alert", "Alert", "ResourceRemoved"],
  "Context": "1234512344512345"
}
```

Is there anything you can do about it?

Reply:
Now I'm treating and removing all duplicated event types in the **POST** operation.

Therefore, if you create a `Subscription` like this:
```javascript
{
  "Destination": "http://www.myserveruri.com",
  "EventTypes": ["Alert", "Alert", "ResourceRemoved"],
  "Context": "1234512344512345"
}
```

The result is going to be exactly like this:

```javascript
{
  "Destination": "http://www.myserveruri.com",
  "EventTypes": ["Alert", "ResourceRemoved"],
  "Context": "1234512344512345"
}
```
So, we no longer have this error when deleting a Subscription.